pkg/controllers/state/suite_test.go

```diff
@@ -779,6 +801,109 @@ var _ = Describe("Inflight Nodes", func() {
 			},
 		}))
 	})
+	It("should only set startup taints once", func() {
```

Comment:
I think we only need the node to populate the startup taints, right? This code to populate the startup taints is meant to cover the case where there is no machine?